Avoid links to ongoing private tournaments (fixes #1691)

ProgramFOX opened the pull request:
When people are in a private tournament, a link to the tournament is still available on their profile, or on the sidebar of an ongoing tournament game, or using the "View tournament" button on TV. This commit makes sure that those links are not displayed for ongoing tournaments. When the tournaments are finished, the links can be displayed because joining is no longer possible, so there's no reason to keep those tournaments private.

ornicar commented on app/templating/TournamentHelper.scala:
```diff
@@ -27,16 +27,26 @@ trait TournamentHelper { self: I18nHelper with DateHelper with UserHelper =>
   }
 
-  def tournamentIdToName(id: String) = tournamentEnv.cached name id getOrElse "Tournament"
+  def tournamentIdToTournament(id: String) = TournamentRepo byId id awaitSeconds 2 get
```
Can't block the thread. Also, that's a DB call in a templating helper.

ornicar commented on modules/api/src/main/RoundApi.scala:
```diff
@@ -136,10 +136,10 @@ private[api] final class RoundApi(
     tourOption.fold(json) { data =>
       json + ("tournament" -> Json.obj(
-        "id" -> data.tour.id,
+        "id" -> (if (hideTourneyIdIfPrivate && data.tour.`private` && !data.tour.isFinished) -1 else data.tour.id),
```
`id: -1` ???

ProgramFOX commented:
@ornicar I have removed the DB call and used tournament caching instead. Those didn't get cached yet so I added this in Cached.scala.

Because the tournaments are cached, I had to replace the .isFinished check with a DateTime comparison: the isFinished value would also get cached. A side effect is that the tournament links are kept private for a bit longer if the tournament ends earlier because people withdrew, but I don't think that this is a huge problem.

Regarding the -1: I tried passing `null` but I got an error then. -1 looked like an acceptable alternative to me.

ProgramFOX commented:
@ornicar Okay, I found a way to get rid of the `-1`: I could use `JsNull` instead.

Please re-review the PR.

ornicar commented:
A better solution would be to reinstate password-protected tournaments.
